examples/api/README.md

```diff
@@ -1,67 +1,125 @@
+system overlay style doesn't make sense on the web (it only changes the
+notification area background color on Android), so you can run the example for
+that on an Android device like so:
+
+```
+% flutter run -d MyAndroidDevice lib/services/system_chrome/system_chrome.set_system_u_i_overlay_style.1.dart
+```
 
 ## Naming
 
 The naming scheme for the files is similar to the hierarchy under
 [packages/flutter/lib/src](../../packages/flutter/lib/src), except that the
-files are represented as directories, and each sample in each file as a separate
-file in that directory. So, for the example above, the examples are from the
+files are represented as directories (without the `.dart` suffix), and each
+sample in the file is a separate file in that directory. So, for the example
+above, where the examples are from the
 [packages/flutter/lib/src/animation/curves.dart](../../packages/flutter/lib/src/animation/curves.dart)
-file, the `Curve2D` class, and the first sample (hence the index "0") for that
-symbol resides in the
-[lib/animation/curves/curve2_d.0.dart](lib/animation/curves/curve2_d.0.dart)
-file.
+file, the `Curve2D` class, the first sample (hence the index "0") for that
+symbol resides in the file named
+[lib/animation/curves/curve2_d.0.dart](lib/animation/curves/curve2_d.0.dart).
+
+Symbol names are converted from "CamelCase" to "snake_case". Dots are left
+between symbol names, so the first example for symbol
+`InputDecoration.prefixIconConstraints` would be converted to
+`input_decoration.prefix_icon_constraints.0.dart`.
+
+If the same example is linked to from multiple symbols, the source will be in
+the canonical location for one of the symbols, and the link in the API docs
+block for the other symbols will point to the first symbol's example location.
 
 ## Authoring
 
-When authoring these examples, place a block like so in the Dartdoc
+> For more detailed information about authoring examples, see [the snippets package](https://pub.dev/packages/snippets).
+
+When authoring examples, first place a block in the Dartdoc
 documentation for the symbol you would like to attach it to. Here's what it
-might look like if you wanted to add a new example to the `Curve2D` class. First
-add the stub to the symbol documentation:
+might look like if you wanted to add a new example to the `Curve2D` class:
 
 ```dart
-/// {@tool dartpad --template=material_scaffold}
+/// {@tool dartpad}
 /// Write a description of the example here. This description will appear in the
 /// API web documentation to introduce the example.
 ///
-/// ```dart
-/// // These are the sections you want to fill out in the template.
-/// // They will be transferred to the example file when you extract it.
-/// ```
+/// ** See code in examples/api/lib/animation/curves/curve2_d.0.dart **
 /// {@end-tool}
 ```
 
-Then install the `extract_sample` command with:
+The "See code" link needs to be formatted exactly as above, with no wrapping or
```

Comment:
It's not really a link (a line maybe?), but that probably will not confuse anyone.